presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -621,15 +623,14 @@ private static void verifyAcidColumn(OrcReader orcReader, int columnIndex, Strin
         List<Integer> desiredColumnsIndex = desiredColumnsByNumber.keySet().stream().collect(Collectors.toList());
         Map<Integer, OrcColumn> fileColumnsByNumber = fileColumns.stream()
                 .collect(toImmutableMap(orcColumn -> orcColumn.getColumnId().getId(), identity()));
-
-        int skipStructColumnCount = 0;
-        HiveColumnHandle handle = null;
-        OrcColumn column = null;
+        for (OrcColumn orcColumn : fileColumns) {
```

Comment:
Why do 2 passes on fileColumns? it can be merged into single pass itself.

Reply:
Made into single pass